snappy_pipeline/workflows/adapter_trimming/__init__.py

```diff
@@ -0,0 +1,334 @@
+            result = {
+                "input": {
+                    "reads_left": list(
+                        sorted(self._collect_reads(wildcards, wildcards.library_name, ""))
+                    )
+                },
+            }
+            reads_right = list(
+                sorted(self._collect_reads(wildcards, wildcards.library_name, "right-"))
+            )
+            if reads_right:
+                result["input"]["reads_right"] = reads_right
+            return result
+
+        assert action == "run", "Unsupported actions"
+        return args_function
+
+    def _collect_reads(self, wildcards, library_name, prefix):
+        """Yield the path to reads
+
+        Yields paths to right reads if prefix=='right-'
+        """
+        _ = library_name
+        folder_name = get_ngs_library_folder_name(self.parent.sheets, wildcards.library_name)
+        pattern_set_keys = ("right",) if prefix.startswith("right-") else ("left",)
+        for _, path_infix, filename in self.path_gen.run(folder_name, pattern_set_keys):
+            yield os.path.join(self.base_path_in, path_infix, filename).format(**wildcards)
+
+
+class BbdukStepPart(AdapterTrimmingStepPart):
+    """bbduk adapter trimming"""
+
+    name = "bbduk"
+
+    def get_resource_usage(self, action):
+        """Get Resource Usage
+
+        :param action: Action (i.e., step) in the workflow, example: 'run'.
+        :type action: str
+
+        :return: Returns ResourceUsage for step.
+
+        :raises UnsupportedActionException: if action not in class defined list of valid actions.
+        """
+        if action not in self.actions:
+            actions_str = ", ".join(self.actions)
+            error_message = f"Action '{action}' is not supported. Valid options: {actions_str}"
+            raise UnsupportedActionException(error_message)
+        return ResourceUsage(
+            threads=self.config["bbduk"]["num_threads"],
+            time="12:00:00",  # 40 hours
+            memory="24000M",
+        )
+
+
+class FastpStepPart(AdapterTrimmingStepPart):
+    """fastp adapter trimming"""
+
+    #: Step name
+    name = "fastp"
+
+    def get_resource_usage(self, action):
+        """Get Resource Usage
+
+        :param action: Action (i.e., step) in the workflow, example: 'run'.
+        :type action: str
+
+        :return: Returns ResourceUsage for step.
+
+        :raises UnsupportedActionException: if action not in class defined list of valid actions.
+        """
+        if action not in self.actions:
+            actions_str = ", ".join(self.actions)
+            error_message = f"Action '{action}' is not supported. Valid options: {actions_str}"
+            raise UnsupportedActionException(error_message)
+        return ResourceUsage(
+            threads=self.config["fastp"]["num_threads"],
+            time="12:00:00",  # 60 hours
+            memory="24000M",
+        )
+
+
+class LinkOutFastqStepPart(LinkOutStepPart):
```

Comment:
Are you actually using any of the methods you inherited from `LinkOutStepPart`? It looks like you could simply go from  `BaseStepPart`.